### ISSUE TITLE
Apollo 1.0 - Codegen - TestMocks - __typename

### ISSUE THREAD

**Issue by jimisaacs** [apollographql/apollo-ios#2357]
## Bug report

When outputting testMocks, all the mocks have the same compiler error, which is:
```
'let' property '__typename' may not be initialized directly; use "self.init(...)" or "self = ..." instead
```

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: `release/1.0`
- Xcode version: 13.4.1
- Swift version: 5
- Package manager: SPM

## Steps to reproduce

Generate test mocks and build a target with them.

**Comment by calvincestari** [comment]
@jimisaacs, could you share the codegen configuration you're using please.

**Comment by jimisaacs** [comment]
@calvincestari 
```swift
let sourceRootURL = FileFinder.findParentFolder()
    .apollo.parentFolderURL()
    .apollo.parentFolderURL()

let inputRootURL = sourceRootURL
    .apollo.childFolderURL(folderName: "MyCodegen GraphQL Sources")

let endpointURL = URL(string: "https://mygraphql.endpoint.com/path/to/graphql")!

let schemaDownload = ApolloSchemaDownloadConfiguration(
   using: .introspection(endpointURL: endpointURL),
   outputPath: inputRootURL.appendingPathComponent("schema.graphqls").path
)

return ApolloCodegenConfiguration(
   schemaName: "ArgoApolloCodeGen",
   input: ApolloCodegenConfiguration.FileInput(
       schemaSearchPaths: [schemaDownload.outputPath],
       operationSearchPaths: [inputRootURL.appendingPathComponent("**/*.graphql").path]
   ),
   output: ApolloCodegenConfiguration.FileOutput(
       schemaTypes: ApolloCodegenConfiguration.SchemaTypesFileOutput(
           path: sourceRootURL
               .apollo.childFolderURL(folderName: "Sources")
               .apollo.childFolderURL(folderName: "MyCodegenTargetDestination")
               .path,
           moduleType: .other
       ),
       operations: .inSchemaModule,
       testMocks: .absolute(
           path: sourceRootURL
               .apollo.childFolderURL(folderName: "Sources")
               .apollo.childFolderURL(folderName: "MyCodegenTestSupportTargetDestination")
               .path
       )
   ),
   schemaDownloadConfiguration: schemaDownload
)
```

**Comment by jimisaacs** [comment]
@calvincestari from what I can gather, I don't think `__typename` should be in the generated `Mock` initializers.

**Comment by calvincestari** [comment]
Thanks @jimisaacs, are you able to share an example of the generated test mocks with `__typename` in it? _Please redact anything you don't want to be public._

**Comment by AnthonyMDev** [comment]
Thanks for the report @jimisaacs! I've figured out the source of this bug and will get it fixed today before we release the 1.0 Beta version this afternoon!

**Comment by AnthonyMDev** [comment]
Fixed in beta 1!
